### DISCUSSION TITLE
Custom origin point

### DISCUSSION THREAD

**Discussion by mattevans-dev** [fabricjs/fabric.js#7313] [Q&A]
Hi,
I was just wondering if it's possible to set a custom origin point? Ideally a numeric value in pixels.
Is this feature currently built in? Reading through the docs it says it only supports 'left', 'top' etc.

Thanks,
Matt

**Comment by asturur** [comment]
it supports left, top etc... and numbers in percentage from 0 to 1.
So for example 0.1 would be 10% from left or from top. it is up to you to calculate that particular pixel at which % correspond
Can you point me at the doc you read so i can update it?

**Reply by mattevans-dev** [reply]
Hi asturur,
Thanks for getting back to me. Is it possible to set the originX and originY outside of the bounds of the object?
E.g this is an example in AE which I am trying to replicate. The anchor point/origin is set to the middle of the canvas and it's outside of the circle object bounds. This allows for more flexibility when it comes to animations:
![fabric-origin-example](https://user-images.githubusercontent.com/55884592/132858488-d7b5a87a-7b24-4562-b17f-a7fe50c59175.gif)

This is where I found it in the docs: http://fabricjs.com/docs/fabric.Object.html#originX
Thanks,
Matt

**Reply by asturur** [reply]
yes you can, but you have to calclulate it.
For example, -5 will be 5 times the width left of the left side, while 4 will be 4 time the width, right of the right side
